src/osd/PrimaryLogPG.cc

```diff
@@ -10593,6 +10593,8 @@ hobject_t PrimaryLogPG::get_fpoid_from_chunk(const hobject_t soid, bufferlist& c
   pg_t raw_pg;
   object_locator_t oloc(soid);
   oloc.pool = pool.info.get_dedup_tier();
+  // check if dedup_tier isn't set
+  ceph_assert(oloc.pool > 0);
```

Comment:
This assert is fine, but the caller should check this and return an error to the client so that we avoid crashing the osd.